open-hackathon/src/hackathon/database/models.py

```diff
@@ -574,4 +574,119 @@ def json(self):
+    online = Column(Integer)  # 0:offline 1:online
+    create_time = Column(DateTime)
+    last_login_time = Column(DateTime)
+
+    def get_admin_user_id(self):
+        return self.id
+
+    def json(self):
+        return to_json(self, self.__class__)
+
+    def __init__(self, **kwargs):
+        super(AdminUser, self).__init__(**kwargs)
+        if self.create_time is None:
+            self.create_time = datetime.utcnow()
+        if self.last_login_time is None:
+            self.last_login_time = datetime.utcnow()
+#        if self.slug is None:
+#            self.slug = str(uuid.uuid1())[0:8]  # todo generate a real slug
+
+
+class AdminEmail(Base):
+    __tablename__ = 'admin_email'
+    id = Column(Integer, primary_key=True)
+    name = Column(String(80))
+    email = Column(String(120))
+    primary_email = Column(Integer)  # 0:NOT Primary Email 1:Primary Email
+    verified = Column(Integer)  # 0 for not verified, 1 for verified
+
+    admin_id = Column(Integer, ForeignKey('admin_user.id', ondelete='CASCADE'))
+    admin = relationship('AdminUser', backref=backref('emails', lazy='dynamic'))
+
+    def __init__(self, **kwargs):
+        super(AdminEmail, self).__init__(**kwargs)
+
+
+
+class AdminToken(Base):
+    __tablename__ = 'admin_token'
+    id = Column(Integer, primary_key=True)
+    token = Column(String(50), unique=True, nullable=False)
+
+    admin_id = Column(Integer, ForeignKey('admin_user.id', ondelete='CASCADE'))
+    admin = relationship('AdminUser', backref=backref('tokens', lazy='dynamic'))
+
+    issue_date = Column(DateTime)
+    expire_date = Column(DateTime, nullable=False)
+
+    def json(self):
+        return to_json(self, self.__class__)
+
+    def __init__(self, token, admin, expire_date, issue_date=None):
+        if issue_date is None:
+            issue_date = datetime.utcnow()
+
+        self.token = token
+        self.admin = admin
+        self.expire_date = expire_date
+        self.issue_date = issue_date
+
+    def __repr__(self):
+        return "AdminToken: " + self.json()
+
+
+
+class AdminGroup(Base):
+    __tablename__ = 'admin_group'
+    id = Column(Integer, primary_key=True)
+    name = Column(String(32))
+    hackathon_id = Column(Integer, nullable=False)
+    create_time = Column(DateTime)
+
+    def json(self):
+        return to_json(self, self.__class__)
+
+    def __init__(self, **kwargs):
+        super(AdminGroup, self).__init__(**kwargs)
+
+    def __repr__(self):
+        return "AdminGroup: " + self.json()
+
+
+
+class AdminUserGroup(Base):
+    __tablename__ = 'admin_user_group'
+    id = Column(Integer, primary_key=True)
+    admin_email = Column(String(120))
+    state = Column(Integer)
+    remarks = Column(String(255))
+    create_time = Column(DateTime)
+
+    group_id = Column(Integer(), ForeignKey('admin_group.id', ondelete='CASCADE'))
+    admin_user_group = relationship('AdminGroup', backref=backref('admin_user_group', lazy='dynamic'))
```

Comment:
and should it be `backref=backref('admin_user_group',...` since it's a one-to-many relationship?